### PR TITLE
Add console socket

### PR DESCRIPTION
**Pull request by saschagrunert**
A few TODOs are still open but the basic fd passing works for now.

### CODE REVIEW

File: conmon-rs/server/src/console.rs [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+                        }
+                        Err(e) => {
+                            return Err(e.into());
+                        }
+                    }
+                }
+            }
+        };
+
+        let (left, right) = tokio::join!(sut.start(), verify);
```
Comment (saschagrunert) [inline conmon-rs/server/src/console.rs:113]: @haircommander is this usable? I mean the whole async can run in parallel, but we would have to join the tasks together and the one from the `Console` is blocking on `await`.

Comment (haircommander) [reply conmon-rs/server/src/console.rs:113]: the goal is to be able to have the container i/o processed in parallel with other i/o processing and container creation. as long as a user's terminal remains responsive, and conmon-rs can make foward progress simultaneously, I think we're good. I am having trouble deciding if this code block does that by looking at it though

Comment (saschagrunert) [reply conmon-rs/server/src/console.rs:113]: I'll do some more experiments and unmark the WIP when ready for review.

**Review by haircommander — CHANGES_REQUESTED** [review]
Now that we actually create a container, wdyt about wiring this up? If you'd prefer as a follow up that's fine with me

**Comment by saschagrunert** [comment]
> Now that we actually create a container, wdyt about wiring this up? If you'd prefer as a follow up that's fine with me

Yeah I can follow up on that as well as those TODOs :)

File: conmon-rs/server/src/console.rs [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+        })
+    }
+
+    async fn handle_fd_receive(
+        mut stream: UnixStream,
+        path: &Path,
+        fd_state: Arc<RwLock<Option<RawFd>>>,
+    ) -> Result<()> {
+        loop {
+            if stream.ready(Interest::READABLE).await?.is_readable() {
```
Comment (haircommander) [inline conmon-rs/server/src/console.rs:94]: what if we made this `if !stream.ready(Interest::READABLE).await?.is_readable() { continue; }` and then we could  move the rest of the block in?

Comment (saschagrunert) [reply conmon-rs/server/src/console.rs]: Sound good, changed as suggested :+1:

File: conmon-rs/server/src/console.rs [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+
+    async fn handle_fd_receive(
+        mut stream: UnixStream,
+        path: &Path,
+        fd_state: Arc<RwLock<Option<RawFd>>>,
+    ) -> Result<()> {
+        loop {
+            if !stream.ready(Interest::READABLE).await?.is_readable() {
+                continue;
+            }
```
Comment (haircommander) [inline conmon-rs/server/src/console.rs:96]: something to think about is whether we can optimize this a bit. obviously we don't want to miss any data from the fd or cause the user to stall, but it seems like we're busy looping here and that may cause unneeded cpu usage

**Comment by haircommander** [comment]
LGTM, needs rebase though..

**Comment by saschagrunert** [comment]
> LGTM, needs rebase though..

Rebased.

**Review by rphillips — APPROVED** [review]

**Pull request merged**